rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportService.java

```diff
@@ -33,7 +44,7 @@ public class SW360ReportService {
     LicenseService.Iface licenseClient = thriftClients.makeLicenseClient();
 
     public ByteBuffer getProjectBuffer(User user, boolean extendedByReleases, String projectId) throws TException {
-        if (projectId != null && validateProject(projectId, user)) {
+        if (projectId != null && !validateProject(projectId, user)) {
```

Comment:
Shouldn't it be "If project is null or it is invalid"?
```suggestion
        if (projectId != null || !validateProject(projectId, user)) {
```

Reply:
If projectId==null, No need to check whether its a valid project or not. If its not a null and then check its a valid project or not, based on that next logic will run.

Reply:
@keerthi-bl Ah, but with current logic, if the `projectId == null` then the first condition becomes false and will not check for `validateProject()` and short-circuits the `if` to false. Thus, no exception will be thrown.

If the condition is rewritten as `if (projectId == null || !validateProject(projectId, user))`, the even if the `projectId == null` or if it is not a valid project, the exception is thrown.

If the exception has to be thrown when the `projectId == null`, then the `if` condition above has to be rewritten.

Reply:
@GMishx :

The method is calling from 2 different places. One to fetch project based on project id and one with based on user. If the method called for user, i no need to consider it for a valid project or not, thus projectId will be null based on that user condition.

Reply:
@GMishx 

As per the discussion, comments added. Let me know, can we mark it as resolved.